src/Illuminate/Database/Query/Builder.php

```diff
@@ -1464,6 +1471,78 @@ protected function restoreFieldsForCount()
 	}
 
 	/**
+	 * Get the count of the total records for the paginator.
+	 *
+	 * @return int
+	 */
+	public function getGroupCountForPagination()
+	{
+		$sql = $this->toSql();
+		$this->backupGroupFieldsForCount();
+
+		$this->from = new Expression("($sql) aggregator_query");
+		$total = $this->count();
+
+		$this->restoreGroupFieldsForCount();
+
+		return $total;
+	}
+
+	/**
+	 * Backup some fields for the pagination count.
+	 *
+	 * @return void
+	 */
+	protected function backupGroupFieldsForCount()
+	{
+		$fields = [
+			'from',
+			'joins',
+			'wheres',
+			'groups',
+			'havings',
+			'orders',
+			'limit',
+			'offset',
+			'unions',
+		];
+
+		foreach ($fields as $field)
+		{
+			$this->backups[$field] = $this->{$field};
+
+			$this->{$field} = null;
+		}
+	}
+
+	/**
+	 * Restore some fields after the pagination count.
+	 *
+	 * @return void
+	 */
+	protected function restoreGroupFieldsForCount()
+	{
+		$fields = [
```

Comment:
I'm just following the format of backupFieldsForCount which also hard-codes them.  Maybe a different issue to decide if those should be extracted or not.